xbmc/utils/URIUtils.cpp

```diff
@@ -263,7 +263,8 @@ void URIUtils::GetCommonPath(std::string& strParent, const std::string& strPath)
 {
   // find the common path of parent and path
   unsigned int j = 1;
-  while (j <= std::min(strParent.size(), strPath.size()) && strnicmp(strParent.c_str(), strPath.c_str(), j) == 0)
+  while (j <= std::min(strParent.size(), strPath.size()) &&
+         StringUtils::CompareNoCase(strParent.c_str(), strPath.c_str(), j) == 0)
```

Comment:
Think this `.c_str()` here and on much other places no more needed.

Would it not better to remove them direct?

In any case, thanks for standardizing this in Kodi!

Reply:
Nice catch!

Reply:
Yep that's a good cleanup. I'll add those in